tests/allschemes/channels008/Main.idr

```diff
@@ -0,0 +1,31 @@
+import System
+import System.Concurrency
+
+-- Simple producing thread.
+producer : Channel Nat -> Nat -> IO ()
+producer c n = ignore $ producer' n
+  where
+    producer' : Nat -> IO ()
+    producer' Z     = pure ()
+    producer' (S n) = do
+      channelPut c n
+      sleep 1
+
+-- Test that channelGetWithTimeout works as expected.
+main : IO ()
+main =
+  do c    <- makeChannel
+     tids <- for [0..11] $ \n => fork $ producer c n
+     vals <- for [0..11] $ \_ => channelGetWithTimeout c 5
+     ignore $ traverse (\t => threadWait t) tids
+     let vals' = map (\val => case val of
+                        Nothing   =>
+                          0
+                        Just val' =>
+                          val'
+                     ) vals
+         s     = sum vals'
```

Comment:
style nit:

lambda can be replaced with partial application of library routine, map with `<$>`.

`sum $ fromMaybe 0 <$> vals`

The whole thing can then be:

`putStrLn $ sum $ fromMaybe 0 <$> vals`

Adjust `expected` case to simply read `55`.

Reply:
@emdash 

This was addressed via commit [6a5129f](https://github.com/idris-lang/Idris2/pull/3435/commits/6a5129ffafc3537450199fb42a5d7b366f495df1).